packages/core/nuxt-theme-module/theme/composables/useUiNotification/index.ts

```diff
@@ -1,17 +1,19 @@
 import { computed, reactive } from '@vue/composition-api';
 
-interface UiNotification {
+interface UseUiNotification {
   message: string;
-  action: { text: string; onClick: Function };
   type: 'danger' | 'success' | 'info';
-  icon: string;
-  persist: boolean;
-  id: symbol;
-  dismiss: () => void;
+  action?: { text: string; onClick: Function };
+  icon?: string;
+  persist?: boolean;
+  id?: symbol;
+  dismiss?: () => void;
 }
 
+type SendUiNotificationParams = Omit<UseUiNotification, 'id'>;
+
 interface Notifications {
-  notifications: Array<UiNotification>;
+  notifications: Array<UseUiNotification>;
```

Comment:
its better not to create types with just 1 property, it reduces readability (just use `Array<UseUiNotification>`)